content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+
+[source]
+----
+a2enmod proxy
+a2enmod proxy_http
+a2enmod headers
+----
+
+A typical set up for mod_proxy would look like this:
+
+[source]
+----
+ProxyPass         /jenkins  http://localhost:8081/jenkins nocanon
+ProxyPassReverse  /jenkins  http://localhost:8081/jenkins
+ProxyRequests     Off
+AllowEncodedSlashes NoDecode
+
+# Local reverse proxy authorization override
+# Most unix distribution deny proxy by default
+# See /etc/apache2/mods-enabled/proxy.conf in Ubuntu
+<Proxy http://localhost:8081/jenkins*>
+  Order deny,allow
+  Allow from all
+</Proxy>
+----
+
+This assumes that you run Jenkins on port 8081.
+
+For this set up to work, the context path of Jenkins *must be the same*
+between your Apache and Jenkins (that is, you can't run Jenkins on
+http://localhost:8081/ci and have it exposed at
+http://localhost:80/jenkins). +
+Set the context path in Windows by modifying the `+jenkins.xml+`
+configuration file and adding --prefix=/jenkins (or similar) to the
+<arguments> entry. +
+Set the context path  when using the Ubuntu package by adding
+`+--prefix=/jenkins+` to `+JENKINS_ARGS+` in `+/etc/default/jenkins+` (
+or in `+/etc/sysconfig/jenkins+` for RHEL/CentOS package)
+
+When running on a dedicated server and you are using / as context, make
+sure you add a slash at the end of all URLs in proxy params in apache.
+Otherwise you might run into proxy errors. So
+
+[source]
+----
+ProxyPass / http://localhost:8080/ nocanon
+----
+
+instead of
+
+[source]
+----
+ProxyPass / http://localhost:8080 nocanon     # wont work
+----
+
+Note that this does *not* apply to the `+ProxyPassMatch+` directive,
+which behaves differently than `+ProxyPass+`.
+Below is an example of `+ProxyPassMatch+` to proxy all URLs other than
+`+/.well-known+` (a URL required by letsencrypt):
+
+[source]
+----
+ProxyPassMatch  ^/(?\!.well-known)  http://localhost:8080 nocanon
+----
+
+The _ProxyRequests Off_ prevents Apache from functioning as a forward
+proxy server (except for _ProxyPass_), it is advised to include it
+unless the server should function as a proxy.
+
+Both the `+nocanon+` option to `+ProxyPass+`, _and_
+`+AllowEncodedSlashes NoDecode+`, are required for certain Jenkins
+features to work.
+
+If you are running Apache on a Security-Enhanced Linux (SE-Linux)
+machine it is essential to make SE-Linux do the right thing by issuing
+as root
+
+[source]
+----
+setsebool -P httpd_can_network_connect true
+----
+
+If this is not issued Apache will not be allowed to forward proxy
+requests to Jenkins and only an error message will be displayed.
+
+Because Jenkins already compress its output, you can not use the normal
+proxy-html filter to modify urls:
+
+[source]
+----
+SetOutputFilter proxy-html
+----
+
+Instead you can use the following:
+
+[source]
+----
+SetOutputFilter INFLATE;proxy-html;DEFLATE
+ProxyHTMLURLMap http://your_server:8080/jenkins /jenkins
+----
+
+http://wiki.uniformserver.com/index.php/Reverse_Proxy_Server_2:_mod_proxy_html_2 +
```

Comment:
I don't understand what is trying to be said here.  This URL opens a page that does not exist on a site that looks somewhat like a wiki.  It either needs to be removed or replaced.

```suggestion
```